rust/kcl-lib/src/execution/kcl_value.rs

```diff
@@ -827,6 +1025,33 @@ impl fmt::Display for RuntimeType {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ArrayLen {
+    None,
```

Comment:
What does None actually represent? Actually, I guess, how could an array's length not be known? Can't we just count the elements and always find its length? Or store the length as a field and always keep it up to date?

Reply:
This would be a static type where the length is not specified, rather than the type of a value